pkg/vm/vm_test.go

```diff
@@ -927,8 +927,81 @@ func TestAssertEqualInstruction(t *testing.T) {
 		require.NoError(t, err)
 		assert.Equal(t, mem.MemoryValueFromInt(-5), mv)
 	})
+
+	t.Run("multiplication", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("[ap] = [ap - 1] * [fp];")
+		setInitialReg(vm, 3, 1, 0)
+
+		writeToDataSegment(vm, vm.Context.Fp, 5)
+		writeToDataSegment(vm, vm.Context.Ap-1, 10)
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		mv, err := vm.Memory.Read(ExecutionSegment, vm.Context.Ap)
+		require.NoError(t, err)
+		assert.Equal(t, mem.MemoryValueFromInt(50), mv)
+	})
+
+	t.Run("division", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("[ap] = [ap + 1] * [fp];")
+		setInitialReg(vm, 3, 1, 0)
+
+		writeToDataSegment(vm, vm.Context.Fp, 2)
+		writeToDataSegment(vm, vm.Context.Ap, 10)
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		mv, err := vm.Memory.Read(ExecutionSegment, vm.Context.Ap+1)
+		require.NoError(t, err)
+		assert.Equal(t, mem.MemoryValueFromInt(5), mv)
+	})
+
 }
 
+// add test for each different form of jump instructions
+func TestJumpInstructions(t *testing.T) {
+
+	// relative jumps
+	t.Run("jmp rel", func(t *testing.T){
+		vm := defaultVirtualMachineWithCode("jmp rel [ap + 1] + [ap - 7];")
+
+		vm.Context.Pc = mem.MemoryAddress{SegmentIndex: 1, Offset: 3}
+		relAddr := uint64(10)
+		res := mem.MemoryValueFromInt(relAddr)
+
+		instruction := a.Instruction{
+			PcUpdate: a.PcUpdateJumpRel,
+		}
+		nextPc, err := vm.updatePc(&instruction, nil, nil, &res)
+
+		require.NoError(t, err)
+		assert.Equal(t, mem.MemoryAddress{SegmentIndex: 1, Offset: 3 + relAddr}, nextPc)
```

Comment:
The test body is not correct and does not match the behaviour of the previous test of the same kind